### PR TITLE
Simplex tree insert duplicated vertices fix vincent

### PR DESCRIPTION
**Pull request by VincentRouvreau**
Fix issue #12

### CODE REVIEW

**Comment by VincentRouvreau** [comment]
Fix issue #12

File: src/Simplex_tree/include/gudhi/Simplex_tree.h [hunk 1]
```diff
@@ -811,19 +814,30 @@ class Simplex_tree {
...
       } else {
-	// FIXME: this interface makes no sense, and it doesn't seem to be tested.
-	insertion_result.first = null_simplex();
+        // FIXME: this interface makes no sense, and it doesn't seem to be tested.
+        insertion_result.first = null_simplex();
       }
     }
-    if (++first == last) return insertion_result;
+    // Next iteration to avoid consecutive equal values
+    while ((vertex_one == *first) && (first < last)) {
```
Comment (mglisse) [inline src/Simplex_tree/include/gudhi/Simplex_tree.h]: At a glance, I would test if `first!=last` before dereferencing `first`, assuming that `*last` is bad.

**Comment by mglisse** [comment]
I think I'd rather call `std::unique` right after `std::sort` in `insert_simplex`/`insert_simplex_and_subfaces` and keep the others as raw, low-level functions that take already sanitized input (so we can add a GUDHI_CHECK `std::adjacent_find` next to `std::is_sorted` if you want). That or reject repetitions.

**Comment by VincentRouvreau** [comment]
Should be ok this way.

File: src/Simplex_tree/include/gudhi/Simplex_tree.h [hunk 2]
```diff
@@ -775,31 +775,26 @@ class Simplex_tree {
...
   }
 
  private:
-  /// Same as insert_simplex_and_subfaces but assumes that the range of vertices is sorted
-  template<class ForwardVertexRange = std::initializer_list<Vertex_handle>>
-  std::pair<Simplex_handle, bool> insert_simplex_and_subfaces_sorted(const ForwardVertexRange& Nsimplex, Filtration_value filt = 0) {
-    auto first = std::begin(Nsimplex);
-    auto last = std::end(Nsimplex);
-    if (first == last)
-      return { null_simplex(), true }; // FIXME: false would make more sense to me.
```
Comment (mglisse) [inline src/Simplex_tree/include/gudhi/Simplex_tree.h]: Can we keep the comment somewhere, saying that `true` is strange?

**Pull request merged**